### PR TITLE
Resolve review comments from Core team.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp [hunk 1]
```diff
@@ -342,7 +342,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
   {
     std::string LastModified;
     std::string Etag;
-    int32_t Quota = int32_t();
+    int64_t Quota = int64_t();
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:345]: this is in GB, right? I think int32 is enough

Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:345]: Why you changed the other to `QuotaInGiB` while keep this unchanged?

**Review by Jinming-Hu — APPROVED** [review]
some minor comments

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:345]: This is generated by swagger and swagger is calling it Quota. It is not visible to customers as well.

Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:345]: But the service does not doc that it is an int32 value. Maybe this SDK is still used 100 years later and Quota exceeds what int32 supports.

**Pull request merged**